sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/InsecureDeserializationTest.cs

```diff
@@ -35,6 +35,13 @@ public class InsecureDeserializationTest
             Verifier.VerifyAnalyzer(@"TestCases\InsecureDeserialization.cs",
                 new InsecureDeserialization(AnalyzerConfiguration.AlwaysEnabled),
                 ParseOptionsHelper.CSharp8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void InsecureDeserialization_Not_Enabled() =>
```

Comment:
can we have such a test for all hotspots? That would get the hotspots via reflection and would find the test files based on the file name and then run this?

otherwise, we'll always going to have the same problem in the future

Reply:
I'm not sure how can we address this. It needs as an input code which breaks the rule. 

We can get indeed all the hotspots and check if the derive from `HotspotDiagnosticAnalyzer` but this is not enough since the test will need noncompliant code specific for the rule or to check if the `IsEnabled` method is called. 

An option would be to add a custom analyzer specific for our project (internal).

Reply:
Discussed offline and we can do this based on the name convention. The test cases files are named the same as the rule class name.